### DISCUSSION TITLE
Ooomp always turning off auto build when starting eclipse

### DISCUSSION THREAD

**Discussion by jcompagner** [eclipse-oomph/oomph#51] [General]
Hi,

When i have installed Oomp then noticed that constantly the auto build was turned off.
this is quite annoying

i see this in the oomph setup.log:

[2023-11-21 10:18:28] Executing startup tasks
[2023-11-21 10:18:29] OpenJDK Runtime Environment 21-ea+25-2212
[2023-11-21 10:18:29] Product self.product.version
[2023-11-21 10:18:29] Workspace C:\workspace__release
[2023-11-21 10:18:29] Bundle org.eclipse.oomph.setup 1.27.0.v20230714-1137, build=172, branch=619f8be9a7f929a284db8759f923f2293a5612e2
[2023-11-21 10:18:29] Bundle org.eclipse.oomph.setup.core 1.28.0.v20230618-0610, build=172, branch=619f8be9a7f929a284db8759f923f2293a5612e2
[2023-11-21 10:18:29] Bundle org.eclipse.oomph.setup.p2 1.20.0.v20230617-1322, build=172, branch=619f8be9a7f929a284db8759f923f2293a5612e2
[2023-11-21 10:18:29] Performing Preference /instance/org.eclipse.core.resources/description.autobuilding = false

when i deinstall i think "Oomph Setup IDE" feature this is gone
Any idea why this happens?

**Comment by merks** [comment]
It appears you have saved a preference to do set it to false!  I assume that's in your User.setup so if you use Navigate -> Open Setup -> User I guess you will find this preference task setting it to false:

![image](https://github.com/eclipse-oomph/oomph/assets/208716/6413afb9-d6c9-4ac9-accb-9fd18e6aa80d)

You can delete that task or you can change the value to true, so that if in some other workspace it's accidentally set of false it will be automatically set to true.

**Comment by jcompagner** [comment]
that's interesting, because i don't have a user.setup (or any *.setup files) in my workspace

I don't really use oomph for anything else then i think to get the Repository viewer view..

so where does it get that info, and why it thinks it needs to do anything in my workspaces (i had this in all of my workspaces)

**Reply by merks** [reply]
Probably reading at least this section will be helpful:

https://wiki.eclipse.org/Eclipse_Oomph_Authoring#Understanding_the_Setup_Engine

**Comment by merks** [comment]
Note the comment about using Navigate -> Open Setup -> User

![image](https://github.com/eclipse-oomph/oomph/assets/208716/d7475cce-627d-497e-a60e-7da88873f87b)

If you use the Show All Resources tool button, you can see where it's physically located:

![image](https://github.com/eclipse-oomph/oomph/assets/208716/49f7b162-f95c-4d17-8c87-6e950bf8a179)

I'll suspect you have lots of things that you'd like set a certain way in your IDEs, for which you could put this to good use.

**Comment by jcompagner** [comment]
ah in the user home dir, i found it yes.
but why is that default false? i never really touched anything of that oomph. 

so now i am just a user that thinks why is that constantly set to false right after start? 

not the mention it s called "setup" so i kind of expect that that runs for a new workspace once, not all the time? or is that logical in this scenario?

**Reply by merks** [reply]
I'm not sure what you mean by default.  I expect at some time you changed the preference and recorded it:

![image](https://github.com/eclipse-oomph/oomph/assets/208716/eef9e9f3-3569-4cee-bf17-f9345177cfa5)

You would have were prompted and then decided saving it would be good:

![image](https://github.com/eclipse-oomph/oomph/assets/208716/734f2207-0453-43ab-943b-7988821ac96d)

And you likely just picked the default policy which is to record it whenever you change it in the future.

Then if you change it in one workspace it's consistent set that way in every workspace on startup.

You might have  not really reviewed the documentation that was presented the first time this happened, perhaps years ago:

![image](https://github.com/eclipse-oomph/oomph/assets/208716/1e44acb4-be2a-4643-9185-24b535f31557)

**Reply by jcompagner** [reply]
ah then that was really by accident at some point
i would never have set that property to false in a workspace of mine

(and i am always just setting it through the windows menu bar under project, if i see it is disabled again..)

So purely from a user endpoint this is a bit hidden and annoying, if i say  through the window menu: enable this, then i just don't expect that eclipse reverts that when i restart...
And i have no idea why that really is, i don't see that anywhere except if i really look in some logs

**Reply by merks** [reply]
This is why the recorder was changed to be disabled by default.  Too many people just didn't read what was written, and didn't think about the implications of their choices.  People just hit buttons to make dialogs go away and there is a risk to doing that.  So now we know better to assume that people will do foolish things.  It remains super handy to have your own personal preferences set the same way everywhere and now you know what to expect.

**Reply by jcompagner** [reply]
and this specific property had another problem. that is that it has a preference setting (that is recorded and stored)
But that same setting you can also do through the menu item, not sure if that is then recorded... 
And that i guess for me was the biggest confusion/annoyance because i did change constantly that setting (through the menu) and every time it was reverted again..

**Reply by merks** [reply]
Indeed  changes through the menu are not recorded.   I imagine that's how most people change it, and then only temporarily.  I do that sometimes and then sometimes I even forget that I've done it only to notice much later after significant confusion that things are rebuilding.  😱
